classes/Carrier.php

```diff
@@ -327,7 +327,7 @@ public function getDeliveryPriceByWeight($total_weight, $id_zone)
      * @param float $total_weight Total weight
      * @param int $id_zone Zone ID
      *
-     * @return float|bool Delivery price, false if not possible
+     * @return bool true if carrier is available
```

Comment:
It's a BC break as the value is now a boolean :thinking:

Reply:
I disagree as the signature doesn't change.

Reply:
sometimes the method returned zero `0`, now it will return false instead, doesn't that make it a breaking change ?

If it's a breaking change, it's minor anyway, since the result is used as a boolean in our codebase anyway

Reply:
@matthieu-rolland I've documented this

Reply:
@Progi1984 It changes the method signature as it wasn't typed.
A signature change when to return type change and break previous behavior.
Type parameters and returns weren't working on PHP5.6, the code is old and wasn't designed with type.

If we do
```php
public static function checkDeliveryPriceByWeight($moduleName, $isStrong, $blablabla)
```

The signature doesn't change, but we don't expect the same parameter, so it's a BC.